src/rules/rule-single-line-max-declarations/index.js

```diff
@@ -0,0 +1,41 @@
+import { isNumber } from "lodash"
+import {
+  isSingleLineString,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "rule-no-single-line"
+
+export const messages = ruleMessages(ruleName, {
+  expected: (quantity) => `Expected a maximum of ${quantity} declaration(s)`,
+})
+
+export default function (quantity) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: quantity,
+      possible: [isNumber],
+    })
+    if (!validOptions) { return }
+
+    root.walkRules(rule => {
+
+      if (!isSingleLineString(rule.toString())) { return }
+
+      const decls = []
+      rule.walkDecls(function (decl) {
+        decls.push(decl)
+      })
```

Comment:
Another possible approach to this:

```js
const decls = rule.nodes.filter(node => node.type === "decl")
```

Current way is fine, works, just thought I'd point that out if it didn't cross you mind already.